間引きの処理を行う関数を編集

seigodev opened the pull request:
#62　create_graphのリファクタリング

seigodev committed:
間引きの処理を行う関数を編集

aabaa commented on demo/create_graph.py:
```diff
@@ -162,42 +159,61 @@ def make_node2ancestors_recursively(node, node2ancestors):
     for target in node.targets:
-        ancestors |= {target}
-        ancestors |= make_node2ancestors_recursively(target, node2ancestors)
+        if target.name != node.name:
```
assert に変更する

aabaa commented on demo/create_graph.py:
```diff
@@ -162,42 +159,61 @@ def make_node2ancestors_recursively(node, node2ancestors):
-
     if not node.targets:
         node2ancestors[node] = set()
         return {node}
```
return {}

aabaa commented on demo/create_graph.py:
```diff
@@ -162,42 +159,61 @@ def make_node2ancestors_recursively(node, node2ancestors):
-    for target in node.targets:
-        if target in all_target_ancestors:
-            removable_dependency_list.append((node, target))
-    return removable_dependency_list
+        if target in removable_targets:
+            removable_edges.append((node, target))
```
```python
removable_targets = removable_targets.intersection(node.targets)
for removable_target in removable_targets:
    removable_edges.append((node, removalbe_target)) 
```

aabaa commented on demo/create_graph.py:
```diff
@@ -162,42 +159,61 @@ def make_node2ancestors_recursively(node, node2ancestors):
+    """
+    removable_edges = list()
+    # node2ancestorsの中からkeyがnode.targets内に存在するもののみ取ってくる。
+    target2ancestors = \
+        {target: ancestors for target, ancestors in node2ancestors.items() if target in node.targets}
+    removable_targets = make_sets_to_union(target2ancestors)
```
```python
ancestors = {}
for target in node.targets:
    ancestors |= node2ancestors[target]
```

aabaa commented on demo/create_graph.py:
```diff
@@ -162,42 +159,61 @@ def make_node2ancestors_recursively(node, node2ancestors):
+
+
+
+def make_sets_to_union(key2sets):
+    """
+    valueに複数のsetを持つ辞書について，それらのsetを一つの合併集合にして返す．
+    Args:
+        key2set: 複数のセットを値に持つ辞書
+    Return:
+        set_union: key2setの値の和集合
+    """
+    set_union = set()
+    for v in key2sets.values():
+        set_union = set_union | v
+    return set_union
 
```
消す

seigodev committed:
Update create_graph.py

aabaa commented on demo/create_graph.py:
```diff
@@ -161,43 +158,47 @@ def make_node2ancestors_recursively(node, node2ancestors):
         node: 全祖先を知りたいノード
         node2ancestors: key=ノード, value=keyの全祖先のセット
     Return:
-        nodeにターゲットが存在しない：要素がnodeのみのセット
-        nodeがnode2ancestors.keys()に存在する:node2ancestors[node]
-        それ以外：nodeの全祖先のノードのセット
+        第1返値：引数nodeの全祖先ノード
+        第2返値：node2ancestors．
```
コメントにnode2ancestorsが関数内で書き換わることを書いておく
第二引数は戻り値から外す

aabaa commented on demo/create_graph.py:
```diff
@@ -161,43 +158,47 @@ def make_node2ancestors_recursively(node, node2ancestors):
-        all_target_ancestors |= node2ancestors[target]
+    waste_edges = list()
+    # node2ancestorsの中からkeyがnode.targets内に存在するもののみ取ってくる。
+    ancestors_after_the_second_generation = set()
```
ancestors_more_than_2_gen_ago

seigodev committed:
Change Requestの修正など

コメントにnode2ancestorsが関数内で書き換わることを書いておく
第二引数は戻り値から外す

ancestors_after_the_second_generation -> ancestors_more_than_2_gen_ago

removableという表現の削除

aabaa commented on demo/create_graph.py:
```diff
@@ -160,44 +157,50 @@ def make_node2ancestors_recursively(node, node2ancestors):
     Args:
         node: 全祖先を知りたいノード
         node2ancestors: key=ノード, value=keyの全祖先のセット
+    Modify:
+        node2ancestors: key=ノード, value=keyの全祖先のセット
```
削除

seigodev committed:
Update create_graph.py
